runtime/lua/vim/lsp/buf.lua

```diff
@@ -143,7 +143,41 @@ function M.document_symbol()
   request('textDocument/documentSymbol', params)
 end
 
+local workspaceFolders = {}
 
+function M.list_workspace_folders()
+  for i, _ in pairs(workspaceFolders) do
+    print(i)
+  end
+end
+
+function M.add_workspace_folder(workspace_folder)
+  workspace_folder = workspace_folder or npcall(vfn.input, "Workspace Folder: ", vfn.expand('%:p:h'))
+  if not (workspace_folder and #workspace_folder > 0) then return end
+  if vim.fn.isdirectory(workspace_folder) == 0 then
+    print(workspace_folder, "is not a valid directory")
```

Comment:
This also prints on the same line, not sure what the best way is to clear the print line and print this on a new line